bin/oneffs.ml

```diff
@@ -49,6 +51,41 @@ let get_info block_size file =
   | None ->
     print_endline "Unset."
 
+let crc_of_file filename =
+  let ic = open_in filename in
+  let finally () = close_in ic in
+  Fun.protect ~finally @@ fun () ->
+  let buf = Bytes.create 0x7ff in
+  let rec go crc = match input ic buf 0 (Bytes.length buf) with
+    | 0 -> crc
+    | len -> go (Checkseum.Crc32.digest_bytes buf 0 len crc)
+    | exception End_of_file -> crc in
+  go Checkseum.Crc32.default
+
+let create block_size oneffs file =
+  let ic = open_in file in
+  let finally () = close_in ic in
+  Fun.protect ~finally @@ fun () ->
+  let length = in_channel_length ic in
+  (* let length = ((ln + (block_size - 1)) / block_size) * block_size in *)
+  let file_crc = crc_of_file file in
+  let hdr = OneFFS.Header.{ length; file_crc } in
+  let sector0 = Cstruct.create block_size in
+  OneFFS.Header.marshal hdr sector0;
+  let oc = open_out oneffs in
+  let finally () = close_out oc in
+  Fun.protect ~finally @@ fun () ->
+  output_string oc (Cstruct.to_string sector0);
+  let buf = Bytes.make block_size '\000' in
+  let rec go () = match input ic buf 0 (Bytes.length buf) with
```

Comment:
Hmm, I don't think this is safe in all cases. This relies on `input` not doing partial reads.

Reply:
yes, we should use `really_input` with an offset :+1:, I will do that.